torch/csrc/jit/codegen/cuda/scheduler.cpp

```diff
@@ -670,6 +874,368 @@ void scheduleReduction(
+        }
+      }
+
+      // 2) Apply heuristics to each reduction
+      std::vector<TensorView*> rfactor_tv;
+      for (auto tv : reduction_tv) {
+        // Reduction Splits
+        // [output-Leftover, Batches-Per-Block, Warp, Loop-Unroll]
+        if (rparams.batches_per_block > 1) {
+          tv->split(0, rparams.batches_per_block);
+        }
+        if (rparams.num_warps > 1) {
+          tv->split(1, rparams.num_warps);
+        }
+        tv->split(-1, rparams.loop_unroll);
+
+        auto reduction_tv_rf = tv->rFactor({-1});
+        rfactor_tv.push_back(reduction_tv_rf);
+      }
+
+      // 3) Split the other TensorViews
+      for (auto tv : other_tv) {
+        if (rparams.batches_per_block > 1) {
+          tv->split(0, rparams.batches_per_block);
+        }
+        if (rparams.num_warps > 1) {
+          tv->split(1, rparams.num_warps);
+        }
+        tv->split(-1, rparams.loop_unroll);
+      }
+
+      // 4) ComputeAt Structure
+      const int kComputeAtAxis = 1;
+      for (auto input : inputs_tv) {
+        for (auto output : outputs_tv) {
+          if (input->getRootDomain().size() == output->getRootDomain().size()) {
+            input->computeAt(output, kComputeAtAxis);
+          }
+        }
+      }
+
+      // 5) Handle Inline-ComputeAt
+      if (rparams.persistent_kernel) {
+        for (const auto input : inputs_tv) {
+          other_tv.push_back(input->cache_after());
+        }
+      } else {
+        auto compute_inline_tv =
+            findTensorViewsToComputeAtInline(fusion, other_tv);
+        for (auto tensor : compute_inline_tv) {
+          auto uses = fusion->unordered_uses(tensor);
+          TORCH_INTERNAL_ASSERT(
+              uses.size() == 1,
+              "This inline-computeAt TensorView ",
+              tensor->name(),
+              " is used multiple times.")
+          Expr* expr = *uses.begin();
+          TensorView* consumer = expr->output(0)->as<TensorView>();
+          tensor->computeAt(consumer, -1);
+        }
+      }
+
+      // 6) Parallel Binding
+      // For all TensorViews
+      for (auto tv : other_tv) {
+        tv->axis(0)->parallelize(ParallelType::BIDx);
+        if (rparams.num_warps > 1) {
+          tv->axis(2)->parallelize(ParallelType::TIDy);
+        }
+        tv->axis(-2)->parallelize(ParallelType::TIDx);
+      }
+
+      // Reduction TensorViews
+      for (auto tv : reduction_tv) {
+        tv->axis(0)->parallelize(ParallelType::BIDx);
+        if (rparams.num_warps > 1) {
+          tv->axis(2)->parallelize(ParallelType::TIDy);
+        }
+        tv->axis(-1)->parallelize(ParallelType::TIDx);
+      }
+
+      // rFactor TensorViews
+      for (auto tv : rfactor_tv) {
+        tv->axis(0)->parallelize(ParallelType::BIDx);
+        if (rparams.num_warps > 1) {
+          tv->axis(2)->parallelize(ParallelType::TIDy);
+        }
+        tv->axis(-2)->parallelize(ParallelType::TIDx);
+      }
+    }
+  } else {
```

Comment:
This else is really far down, can you just comment this is the non-fastest-dim logic?